x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -485,8 +564,56 @@ private static RoleDescriptor.IndicesPrivileges[] parseIndices(String roleName,
         return privileges.toArray(new IndicesPrivileges[privileges.size()]);
     }
 
-    private static RoleDescriptor.IndicesPrivileges parseIndex(String roleName, XContentParser parser, boolean allow2xFormat)
+    private static IndicesPrivileges parseIndex(final String roleName, final XContentParser parser, final boolean allow2xFormat)
```

Comment:
Unfortunately no way to omit it for `parseIndex` but I did omit it for `parseRemoteIndex` as per @ywangd's suggestion in a separate comment. 

Just for context, `allow2xFormat` is there because we used a different format for field-level security definitions in the past. Most callers of the `parse` method do not allow legacy format, but `FileRolesStore` does since we don't want to fail parsing roles defined in files, even when legacy.

Reply:
We really should remove it and there is an open issue for it https://github.com/elastic/elasticsearch/issues/30104 (and 2 old PRs ...)
But it does not have to be part of this PR. Something for tech debt week? :)

Reply:
Yup, I still haven't signed up for any issues other than test failures for tech debt week. I'll pick this one up (and create a Jira issue if needed)